e2e/test/scenarios/sharing/public-dashboard.cy.spec.js

```diff
@@ -286,6 +288,54 @@ describe("scenarios > public > dashboard", () => {
     cy.findByText("Registerkarte als PDF exportieren").should("be.visible");
     cy.url().should("include", "locale=de");
   });
+
+  it("should respect click behavior", () => {
+    createDashboardWithQuestions({
+      dashboardName: "test click behavior",
+      questions: [
+        {
+          name: "orders",
+          query: {
+            "source-table": ORDERS_ID,
+            limit: 5,
+          },
+        },
+      ],
+      cards: [
+        {
+          visualization_settings: {
+            column_settings: {
+              '["name","TOTAL"]': {
+                click_behavior: {
+                  type: "link",
+                  linkType: "url",
+                  linkTemplate: "https://metabase.com",
+                },
+              },
+            },
+          },
+        },
+      ],
+    }).then(({ dashboard }) => {
+      visitPublicDashboard(dashboard.id);
+    });
+
+    // This is a hacky way to intercept the link click we create an a element
+    // with href on fly and remove it afterwards in lib/dom.js
+    cy.window().then(win => {
```

Comment:
Since this based on the implementation of `openInBlankWindow`/`clickLink`, should we spy directly on that?

If we change how the external link is handled in the react code (ie: we actually render a `a` tag), the test will fail in both solutions but at least if we spy on `openInBlankWindow` it's clearer why/what we're doing.

If we instead change only the implementation of `openInBlankWindow` then it will still pass

Reply:
I've tried importing dom/lib in cypress test and got an error about the undefined process. I can spend some time on fixing it but to me both solutions more or less hacky and similar. 

If you can share how can I spy on that openInBlankWindow function in cypress test, it would be really nice

Reply:
I have also tried putting a function on window

```
if (isCypressActive) {
  if (!window.__domUtils) {
    window.__domUtils = {};
  }

  window.__domUtils.openInBlankWindow = openInBlankWindow;
  window.__domUtils.open = open;
}
```

but cypress wasn't able to track function calls unfortunately